tests/test_misc.py

```diff
@@ -281,6 +281,100 @@ def test_listpresignedtransactions(revault_network):
     assert man_res["unvault"]["hex"] is not None
 
 
+@pytest.mark.skipif(not POSTGRES_IS_SETUP, reason="Needs Postgres for servers db")
+def test_listspendtxs(revault_network, bitcoind):
+    rn = revault_network
+    rn.deploy(n_stakeholders=2, n_managers=2, n_stkmanagers=0, csv=5)
+
+    vaults = rn.fundmany([1, 2])
+    for v in vaults:
+        rn.secure_vault(v)
+        rn.activate_vault(v)
+
+    destinations, feerate = rn._any_spend_data(vaults)
+    man = rn.man(0)
+    deposits = []
+    deriv_indexes = []
+    for v in vaults:
+        deposits.append(f"{v['txid']}:{v['vout']}")
+        deriv_indexes.append(v["derivation_index"])
+    man.wait_for_active_vaults(deposits)
+    spend_tx = man.rpc.getspendtx(deposits, destinations, feerate)["spend_tx"]
+
+    for man in rn.mans():
+        spend_tx = man.man_keychain.sign_spend_psbt(spend_tx, deriv_indexes)
+        man.rpc.updatespendtx(spend_tx)
+        wait_for(
+            lambda: len(man.rpc.listspendtxs("non_final")["spend_txs"]) == 1
+        )
+
+    spend_psbt = serializations.PSBT()
+    spend_psbt.deserialize(spend_tx)
+    spend_psbt.tx.calc_sha256()
+    man.rpc.setspendtx(spend_psbt.tx.hash)
+
+    for w in rn.participants():
+        wait_for(
+            lambda: len(w.rpc.listvaults(["unvaulting"], deposits)["vaults"])
+            == len(deposits)
+        )
+
+    # The initiator will see the spend as immature
+    wait_for(
+        lambda: len(man.rpc.listspendtxs("immature")["spend_txs"]) == 1,
+    )
+
+    rn.bitcoind.generate_block(4, wait_for_mempool=len(deposits))
+
+    for w in rn.participants():
+        wait_for(
+            lambda: len(w.rpc.listvaults(["unvaulted"], deposits)["vaults"])
+            == len(deposits)
+        )
+
+    # Still immature...
+    wait_for(
+        lambda: len(man.rpc.listspendtxs("immature")["spend_txs"]) == 1
+    )
+
+    rn.bitcoind.generate_block(1)
+
+    for w in rn.participants():
+        wait_for(
+            lambda: len(w.rpc.listvaults(["spending"], deposits)["vaults"])
+            == len(deposits),
+        )
+
+    # Broadcasted!
+    wait_for(
+        lambda: len(man.rpc.listspendtxs("broadcasted")["spend_txs"]) == 1
+    )
+
+    rn.bitcoind.generate_block(1, wait_for_mempool=[spend_psbt.tx.hash])
+
+    for w in rn.participants():
+        wait_for(
+            lambda: len(w.rpc.listvaults(["spent"], deposits)["vaults"])
+            == len(deposits),
+        )
+
+    wait_for(
+        lambda: len(man.rpc.listspendtxs("broadcasted")["spend_txs"]) == 1
+    )
+
+    vaults = rn.fundmany([3, 4, 5])
+    for v in vaults:
+        rn.secure_vault(v)
+        rn.activate_vault(v)
+    rn.unvault_vaults_anyhow(vaults)
+    rn.cancel_vault(vaults[0])
+
+    wait_for(
+        lambda: len(man.rpc.listspendtxs("broadcasted")["spend_txs"]) == 1,
+        debug_fn=lambda: man.rpc.listspendtxs()
+    )
```

Comment:
Nice to test this. Should be an assert though (and the debug_fn is probably a debugging leftover :p)